File: 250328/이상한 폭탄 2/strange-bomb-2.py
N, K = map(int, input().split())
boombs = [int(input()) for _ in range(N)]

boom = 0
for i in range(N):
    for j in range(i + 1, min(i + K + 1, N)):
        if boombs[i] == boombs[j]:
            boom = max(boom, boombs[i])

print(boom)

            

File: 250328/이상한 폭탄 2/strange-bomb-2.cpp
#include <iostream>

using namespace std;

int N, K;
int boombs[100];

int main() {
    cin >> N >> K;

    for (int i = 0; i < N; i++) {
        cin >> boombs[i];
    }

    int boom = -1;
    for (int i = 0; i < N; i++) {
        for (int j = 1; j <= K; j++) {
            if (boombs[i] == boombs[i + j]) {
                boom = max(boom, boombs[i]);
            }
        }
    }

    cout << boom;
    return 0;
}